site_scons/site_tools/roc/config.py

```diff
@@ -290,6 +295,67 @@ def FindConfigGuess(context):
     context.Result('not found')
     return False
 
+def FindPkgConfigPath(context):
+    env = context.env
+
+    context.Message("Searching PKG_CONFIG_PATH...")
+
+    if env.HasArg('PKG_CONFIG_PATH'):
+        context.Result(env['PKG_CONFIG_PATH'])
+        return True
+
+    if env.HasArg('PKG_CONFIG'):
+        pkg_config = env['PKG_CONFIG']
+    elif env.Which('pkg-config'):
+        pkg_config = 'pkg-config'
+    else:
+        context.Result('not found')
+        return False
+
+    pkg_config_paths = env.CommandOutput('%s --variable pc_path pkg-config' % quote(pkg_config))
+    try:
+        for path in pkg_config_paths.split(':'):
+            if os.path.isdir(path):
+                env['PKG_CONFIG_PATH'] = path
+                break
+    except:
+        pass
+    finally:
+        if 'PKG_CONFIG_PATH' not in env.Dictionary():
+            # https://linux.die.net/man/1/pkg-config the default is libdir/pkgconfig
+            env['PKG_CONFIG_PATH'] = os.path.join(env['ROC_SYSTEM_LIBDIR'], 'pkgconfig')
+            if not os.path.isdir(env['PKG_CONFIG_PATH']):
+                os.makedirs(env['PKG_CONFIG_PATH'])
+        context.Result(env['PKG_CONFIG_PATH'])
+    return True
+
+def AddPkgConfigDependency(context, package, flags):
+    env = context.env
+    if 'PKG_CONFIG_DEPS' not in env.Dictionary():
+        env['PKG_CONFIG_DEPS'] = []
+
+    if env.HasArg('PKG_CONFIG'):
+        pkg_config = env['PKG_CONFIG']
+    elif env.Which('pkg-config'):
+        pkg_config = 'pkg-config'
+    else:
+        return False
+
+    try:
+        env.ParseConfig('%s %s %s' % (quote(pkg_config), package, flags))
+        env.AppendUnique(PKG_CONFIG_DEPS=[package])
+        return True
+    except:
+        return False
+
+def AddPkgConfigLibs(context, libs):
```

Comment:
We can make this a method of `env` instead of a method of `conf` because AddPkgConfigLibs() does not perform any configuration (e.g. searching for headers, etc) and is usually used when we don't have `conf`.

Also, we can extract GeneratePkgConfig and AddPkgConfigLibs into a new file `pkgconfig.py`.